Rename ActionDispatch::IntegrationTest into ActionDispatch::IntegrationTestCase

pschambacher opened the pull request:
### Summary

At the moment we have `ActiveSupport::TestCase`, `ActionCable::TestCase`, `ActionCable::TestCase`, `ActionView::TestCase` and finally `ActionDispatch::SystemTestCase`. It feels like `ActionDispatch::IntegrationTest` was not respecting the convention of other tests. Now it is.

pschambacher committed:
Rename ActionDispatch::IntegrationTest into ActionDispatch::IntegrationTestCase

pschambacher commented:
This PR is dedicated to @eileencodes who pointed this out during her talk.

eileencodes commented on actionpack/lib/action_dispatch/testing/integration.rb:
```diff
@@ -648,4 +648,16 @@ def document_root_element
+        Please use ActionDispatch::IntegrationTestCase adapter instead.
+      MSG
+
+      super
```
This is great and I'm glad you opened this PR! I think that we should move `IntegrationTestCase` out of `Integration` and into `integration_test_case.rb` and then move the testing directory to `integration_testing` so that those match how system tests work now (top level `SystemTestCase` and directory of dependencies in`system_testing`.

pschambacher replied:
Will do!

eileencodes reviewed (changes requested):
This is great! Thanks for working on it. 😄 I left a comment mainly about how I think we should be following the pattern of system tests that I added. Additionally there are some failing railties tests.

pschambacher commented:
Damn railties test! I'll see what's failing and fix them. I had to stop my local run to checkout and head to the airport :-/

pschambacher committed:
Move ActionDispatch::IntegrationTestCase into its own file

pschambacher commented:
@eileencodes What about all the tests in this file? https://github.com/pschambacher/rails/blob/master/actionpack/test/controller/integration_test.rb

Do we want to have each of these in an individual test file within `integration_testing` or should I just get the `IntegrationTestCaseTest` out into its own file?

pschambacher commented:
My own 2 cents is that it would be easier to have them in separate files. Having smaller test files would be easier to read, etc. I'm asking because if that's gonna be rejected I'll save myself from doing all that.

pschambacher committed:
Separate integration file into integration_testing modules

pschambacher commented:
@eileencodes OK the build is green. Let me know if I need to change anything, in particular about this large test file 😄

pschambacher commented:
Rebased on master and squashed the smaller commits. I think it's better to keep the 3 remaining commits separate to keep a clearer history but if you prefer I can squash into a single commit.

eileencodes commented on actionpack/lib/action_dispatch.rb:
```diff
@@ -91,7 +91,7 @@ module Session
     autoload :Assertions
     autoload :Integration
-    autoload :IntegrationTest, "action_dispatch/testing/integration"
+    autoload :IntegrationTestCase, "action_dispatch/integration_test_case"
```
This needs to autload both `IntegrationTest` and `IntegrationTestCase`

eileencodes commented on actionpack/lib/action_dispatch/integration_test.rb:
```diff
@@ -0,0 +1,13 @@
+      MSG
+
+      super
+    end
```
I was chatting with @matthewd and he mentioned that Minitest has since changed their internal naming from using `TestCast` to `Test`.

Can you update this PR to make `IntegrationTestCase` an alias of `IntegrationTest`? That way we don't force users to change, but we can add internal consistency to our test cases?

pschambacher replied:
I'll do that.

eileencodes reviewed (changes requested):
Hey @pschambacher sorry for the delay. I have a couple more changes I'd like to see before merging this.

rafaelfranca commented on actionpack/lib/action_dispatch/testing/integration.rb:
```diff
@@ -1,651 +0,0 @@
-      end
-    end
-
-    module Behavior
```
This is going to break RSpec and cause some trouble for people upgrading their applications. Is all this trouble worth given this name was added even before Rails 3.2?

matthewd replied:
IMO AD::IntegrationTest and AD::SystemTestCase must be reconciled... and I think it's worth aligning them with the various *::TestCase names. (I could be swayed to SystemTest + *::TestCase, or a more thorough SystemTest + *::Test.)

I think any such renaming would involve silent aliases for any displaced names, so unless RSpec cares about the actual `name` of these modules/classes, it should be fine either way.

pschambacher replied:
I tend to like consistency. I do not exactly care if things are named `XXXTest` or `XXXTestCase` or `XXX::Test(Case?)` but I'd like them to be consistent 😄

rafaelfranca replied:
Yeah, I'm fine with silent aliases.
